### PR TITLE
Port CI process to Github Actions

### PR DESCRIPTION
**Pull request by brandongregoryscott**
See AndcultureCode/AndcultureCode#64 - Investigate alternatives to TravisCI 

Replaces the current Travis configuration with a Github Actions configuration. This configuration additionally runs on macOS, Windows, and two versions of node + dotnet. While Github Actions are currently unlimited for public repositories, we have the ability to exclude certain jobs from the matrix that are probably overkill for anyone forking + developing. These will still run on pull request to ensure compatibility across OS/runtime versions.

One step that we'll need to follow up with is creating a bot integration + adding the slack bot token to the secrets section in this repository. See https://github.com/voxmedia/github-action-slack-notify-build/#setup

-   [x] Related GitHub issue(s) linked in PR description
-   [x] Destination branch merged, built and tested with your changes
-   [x] Code formatted and follows best practices and patterns
-   [x] Code builds cleanly (no _additional_ warnings or errors)
-   [x] Manually tested
-   [x] Automated tests are passing
-   [x] No _decreases_ in automated test coverage
-   [-] Documentation updated (readme, docs, comments, etc.)
-   [-] Localization: No hard-coded error messages in code files (_minimally_ in string constants)

### CODE REVIEW

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -0,0 +1,105 @@
+name: AndcultureCode.Cli
```
Comment (wintondeshong) [inline .github/workflows/main.yaml:1]: Dig it! I'll have to look at the documentation some time to see what all is available. Looks pretty nice with matrix builds

**Review by wintondeshong — APPROVED** [review]
Thanks for investigating! Sounds like we will need to go down the line and update our repos

File: .github/workflows/main.yaml (hunk 1 above)
Comment (brandongregoryscott) [reply .github/workflows/main.yaml:1]: For sure - The way the actions are setup makes it super easy to plug & play different configurations in a single job, which was more of a pain in Travis (could only use one main image as a base, and install any other dependencies manually..). Travis had matrix builds, too, but you'd run out your free credits real quickly with any sort of advanced matrix setup.

**Comment by brandongregoryscott** [comment]
> Thanks for investigating! Sounds like we will need to go down the line and update our repos

Yeah, not urgent but the sooner we have them all aligned the better. We might be able to publish a base configuration that we could share across all of our C# repos which could ease the transition - I'll have to investigate that further.

**Comment by brandongregoryscott** [comment]
@all-contributors please add @brandongregoryscott for  code, doc, infra, maintenance, tests,

**Pull request merged**